versions/draft-3/SPEC.md

```diff
@@ -636,6 +636,16 @@ $document = ($import | $task | $workflow)+
 
 `$document` is the root of the parse tree and it consists of one or more import statement, task, or workflow definition
 
+## Versioning
+
+For portability purposes it is critical that WDL documents be versioned so an engine knows how to process it. A WDL document is versioned with a `version` statement, for eample
```

Comment:
where must this line appear?

Reply:
also `eample`->`example`

Reply:
each spec should explicitly state its own version (not as an example). So the sentance should read, wdl-files adhering to the current version of this spec must have `version draft-3` as the first line in the file.

Reply:
I'd make it a comment and include the term "WDL" in it...so the version line I'd rally behind would be
```
#WDL version draft-3
```
That way if you happen to write draft-2 compliant code you are able to run it with an old engine.

Reply:
I disagree w/ it being a comment, I think it should be a first class construct in the WDL syntax. That way it can be easily parsed without having to special case that specific comment.

re the spec vs example thing, my intention was that e.g. the next version would state that allowable values are `draft-3` and `whateverIsNext`. Perhaps I could start that trend with an N of 1 here?

Reply:
As to "where" I think technically the answer will be "anywhere" (again, going back to making the parsing reasonable), but that best practices would be to put it at the top.

Reply:
I disagree with your disagreement :-) 

it needs to be special cased anyway, since you don't know what parser you need to use...the only way to protect against future changes to the spec and still be able to read any file and tell if it's a spec you support is to know EXACTLY where to find the version. thus: first line in the file. we are already talking about changing the block delimiter...how will your engine find the version if in draft 5 a wdl is unrecognizable to a draft-3 parser?

Reply:
"whatever is next" requires that we move off of the english language in "draft-3" and moe to something computable...how about 0.3 ? then you have a major.minor.point versions and you can be very clear about which version is greater than which

Reply:
The intention is that "whateverIsNext" is called "1.0.0" ;)

The sole reason for the existence of `draft-3` was to provide the ability to right some wrongs from `draft-2` prior to `1.0.0` while also allowing for the ability for backwards incompatible changes if necessary.

We (advisory board) did/do not want this period to go on forever, we've just been bad about driving the changes we felt necessary in a timely manner. This was one of them.

Reply:
**That** is out of the scope of this PR. If you want to change the versioning scheme, feel free to make a separate proposal.

Reply:
You still haven't responded to my concern about parsability...

consider that I might have the phrase `version blah` in many places in the wdl, without meaning to set the version (for example, we have a line that is 
```
version = version
```
in one of the wdl files right now in dsde-pipelines. Without actually parsing the file, you cannot tell if that is part of a command, part of a task, part of a workflow or part of a comment (maybe in the future we want to allow comment blocks?) and without knowing the version you cannot parse the file...so my sense is that by saying "the version line can be anywhere") you will actually limit the spec rather than enable breaking changes which is what the version is support to allow.

Reply:
It's because once in a while you make good points and I need to think about them for a bit :P

Reply:
Sure, I just didn't want it to drown in the other less important comments.

Reply:
Nope, I was going to take another pass through here - hopefully later today. Also part of me was hoping that other people would start chiming in to give other food for thought and/or let me read the direction the wind is blowing instead of us two just going back and forth. :)